DataFormats/L1TCalorimeterPhase2/interface/DigitizedClusterCorrelator.h

```diff
@@ -0,0 +1,164 @@
+                               unsigned int hoe,
+                               bool iso,
+                               unsigned int fb,
+                               unsigned int timing,
+                               bool shape,
+                               unsigned int brems,
+                               int iGCTCard) {
+      clusterData = (((ap_uint<64>)digitizePt(pt_f)) | ((ap_uint<64>)digitizeIEta(iEta) << 12) |
+                     ((ap_uint<64>)digitizeIPhi(iPhi) << 18) | ((ap_uint<64>)digitizeIEtaCr(iEtaCr) << 23) |
+                     ((ap_uint<64>)digitizeIPhiCr(iPhiCr) << 26) | ((ap_uint<64>)digitizeHoE(hoe) << 29) |
+                     ((ap_uint<64>)digitizeIso(iso) << 33) | ((ap_uint<64>)digitizeFb(fb) << 36) |
+                     ((ap_uint<64>)digitizeTiming(timing) << 42) | ((ap_uint<64>)digitizeShape(shape) << 47) |
+                     ((ap_uint<64>)digitizeBrems(brems) << 48));
+      idxGCTCard = iGCTCard;
+    }
+
+    ap_uint<64> data() const { return clusterData; }
+
+    // Other getters
+    float ptLSB() const { return LSB_PT; }
+    ap_uint<12> pt() const { return (clusterData & 0xFFF); }
+    ap_uint<6> eta() const { return ((clusterData >> 12) & 0x3F); }   // (six 1's) 0b111111 = 0x3F
+    ap_uint<5> phi() const { return ((clusterData >> 18) & 0x1F); }   // (five 1's) 0b11111 = 0x1F
+    ap_uint<3> etaCr() const { return ((clusterData >> 23) & 0x7); }  // (three 1's) 0b111 = 0x7
+    ap_uint<3> phiCr() const { return ((clusterData >> 26) & 0x7); }
+    ap_uint<4> hoe() const { return ((clusterData >> 29) & 0xF); }  // (four 1's) 0b1111 = 0xF
+    ap_uint<3> iso() const { return ((clusterData >> 33) & 0x7); }
+    ap_uint<6> fb() const { return ((clusterData >> 36) & 0x3F); }
+    ap_uint<5> timing() const { return ((clusterData >> 42) & 0x1F); }
+    ap_uint<1> shape() const { return ((clusterData >> 47) & 0x1); }
+    ap_uint<1> brems() const { return ((clusterData >> 48) & 0x1); }
+    unsigned int cardNumber() const { return idxGCTCard; }  // which GCT card (0, 1, or 2)
+
+    const int unusedBitsStart() const { return 49; }  // unused bits start at bit 49
+
+    // Prints
+    void print(const std::string location);
+    void printPt(void) const { std::cout << "pt: " << std::bitset<12>{pt()} << std::endl; };
+    void printPtFloat(void) const { std::cout << "pt: " << pt() * ptLSB() << std::endl; };
+    void printEta(void) const { std::cout << "eta: " << std::bitset<6>{eta()} << std::endl; };
+    void printPhi(void) const { std::cout << "phi: " << std::bitset<5>{phi()} << std::endl; };
+    void printEtaCr(void) const { std::cout << "etaCr: " << std::bitset<3>{etaCr()} << std::endl; };
+    void printPhiCr(void) const { std::cout << "phiCr: " << std::bitset<3>{phiCr()} << std::endl; };
+    void printHoE(void) const { std::cout << "hoe: " << std::bitset<4>{hoe()} << std::endl; };
+    void printCardNumber(void) const { std::cout << "GCT card number: " << cardNumber() << std::endl; };
```

Comment:
These are never going to be acceptable in CMSSW code. Please remove them.

Reply:
Removed as of [e7e753d](https://github.com/cms-l1t-offline/cmssw/pull/1069/commits/e7e753df69e7aff845a3b3dfbaed9a06c9fdaec6)